homeassistant/components/media_player/monoprice.py

```diff
@@ -0,0 +1,183 @@
+"""
+Support for interfacing with Monoprice 6 zone home audio controller.
+
+https://www.monoprice.com/product?p_id=10761
+"""
+import logging
+
+import voluptuous as vol
+from homeassistant.components.media_player import (
+    SUPPORT_TURN_ON, SUPPORT_TURN_OFF, SUPPORT_VOLUME_MUTE,
+    SUPPORT_VOLUME_SET, SUPPORT_VOLUME_STEP, SUPPORT_SELECT_SOURCE,
+    MediaPlayerDevice, PLATFORM_SCHEMA)
+from homeassistant.const import (CONF_PORT, STATE_OFF, STATE_ON, CONF_NAME)
+import homeassistant.helpers.config_validation as cv
+
+
+REQUIREMENTS = ['pymonoprice==0.2']
+
+_LOGGER = logging.getLogger(__name__)
+
+SUPPORT_MONOPRICE = SUPPORT_VOLUME_MUTE | SUPPORT_VOLUME_SET | \
+                    SUPPORT_VOLUME_STEP | SUPPORT_TURN_ON | \
+                    SUPPORT_TURN_OFF | SUPPORT_SELECT_SOURCE
+
+ZONE_SCHEMA = vol.Schema({
+    vol.Required(CONF_NAME): cv.string,
+})
+
+SOURCE_SCHEMA = vol.Schema({
+    vol.Required(CONF_NAME): cv.string,
+})
+
+CONF_ZONES = 'zones'
+CONF_SOURCES = 'sources'
+
+# Valid zone ids: 11-16 or 21-26 or 31-36
+ZONE_IDS = vol.All(vol.Coerce(int), vol.Any(vol.Range(min=11, max=16),
+                                            vol.Range(min=21, max=26),
+                                            vol.Range(min=31, max=36)))
+
+# Valid source ids: 1-6
+SOURCE_IDS = vol.All(vol.Coerce(int), vol.Range(min=1, max=6))
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_PORT): cv.string,
+    vol.Required(CONF_ZONES): vol.Schema({ZONE_IDS: ZONE_SCHEMA}),
+    vol.Required(CONF_SOURCES): vol.Schema({SOURCE_IDS: SOURCE_SCHEMA}),
+})
+
+
+# pylint: disable=unused-argument
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the Monoprice 6-zone amplifier platform."""
+    port = config.get(CONF_PORT)
+
+    if port is None:
+        _LOGGER.error("Invalid config. Expected %s", CONF_PORT)
+        return False
+
+    from pymonoprice import Monoprice
+    monoprice = Monoprice(port)
```

Comment:
What happens if the device is not available? I guess that there will be an exception.

Reply:
@fabaff Yes, a `serial.SerialException` will be thrown

Reply:
You need to catch the exception. We don't want to fill the log with tracebacks.

Reply:
done